extensions/dicom-seg/src/components/SegmentationSettings/SegmentationSettings.js

```diff
@@ -0,0 +1,169 @@
+              showPercentage
+              step={1}
+              min={0}
+              max={100}
+              onChange={event => save('outlineAlpha', toFloat(event.target.value))}
+            />
+            <CustomRange
+              value={state.outlineWidth}
+              label="Width"
+              showValue
+              step={1}
+              min={0}
+              max={5}
+              onChange={event => save('outlineWidth', parseInt(event.target.value))}
+            />
+          </>
+        )}
+      </div>
+      {(state.renderFill || state.renderOutline) && (
+        <div
+          className="settings-group"
+          style={{ marginBottom: state.shouldRenderInactiveLabelmaps ? 15 : 0 }}
+        >
+          <CustomCheck
+            label="Render inactive segmentations"
+            checked={state.shouldRenderInactiveLabelmaps}
+            onChange={() => check('shouldRenderInactiveLabelmaps')}
+          />
+          {state.shouldRenderInactiveLabelmaps && (
+            <>
+              {state.renderFill && (
+                <CustomRange
+                  label="Fill Opacity"
+                  showPercentage
+                  step={1}
+                  min={0}
+                  max={100}
+                  value={state.fillAlphaInactive * 100}
+                  onChange={event => save('fillAlphaInactive', toFloat(event.target.value))}
+                />
+              )}
+              {state.renderOutline && (
+                <CustomRange
+                  label="Outline Opacity"
+                  showPercentage
+                  step={1}
+                  min={0}
+                  max={100}
+                  value={state.outlineAlphaInactive * 100}
+                  onChange={event => save('outlineAlphaInactive', toFloat(event.target.value))}
+                />
+              )}
+            </>
+          )}
+        </div>
+      )}
+    </div>
+  );
+};
+
+const CustomCheck = ({ label, checked, onChange }) => {
+  return (
+    <div className="custom-check">
+      <label>
+        <span>{label}</span>
+        <input type="checkbox" checked={checked} onChange={onChange} />
+      </label>
+    </div>
+  );
+};
+
+const CustomRange = props => {
+  const { label, onChange } = props;
+  return (
+    <div className="range">
+      <label htmlFor="range">{label}</label>
+      <Range
+        {...props}
+        onChange={event => {
+          event.persist();
+          onChange(event);
+        }}
+      />
+    </div>
+  );
+};
+
+SegmentationSettings.propTypes = {
+  configuration: PropTypes.shape({
+    renderFill: PropTypes.bool,
+    renderOutline: PropTypes.bool,
+    shouldRenderInactiveLabelmaps: PropTypes.bool,
+    fillAlpha: PropTypes.oneOfType([PropTypes.string, PropTypes.number]), /* TODO: why fillAlpha is string? */
+    outlineAlpha: PropTypes.number,
+    outlineWidth: PropTypes.number,
+    fillAlphaInactive: PropTypes.number,
+    outlineAlphaInactive: PropTypes.number,
```

Comment:
Are each of these keys required or optional?

Reply:
Required, updated, thanks.